src/rgw/rgw_redis_common.cc

```diff
@@ -0,0 +1,54 @@
+#include "rgw_redis_common.h"
+
+#include <iostream>
+
+#include "rgw_redis_scripts.h"
+
+namespace rgw {
+namespace redis {
+
+using boost::redis::config;
+using boost::redis::connection;
+
+int loadLuaFunctions(boost::asio::io_context& io, connection* conn, config* cfg,
+                     optional_yield y) {
+  conn->async_run(*cfg, {}, boost::asio::detached);
+
+  boost::redis::request req;
+  boost::redis::response<std::string> resp;
+  boost::system::error_code ec;
+
+  req.push("FUNCTION", "LOAD", "REPLACE", RGW_LUA_SCRIPT);
+  rgw::redis::redis_exec(conn, ec, req, resp, y);
+
+  if (ec) {
+    std::cerr << "EC Message: " << ec.message() << std::endl;
+    return ec.value();
+  }
+  if (std::get<0>(resp).value() != "rgwlib") return -EINVAL;
+
+  return 0;
+}
+
+RedisResponse doRedisFunc(connection* conn, boost::redis::request& req,
+                          RedisResponseMap& resp, std::string func_name,
```

Comment:
why do you need the `RedisResponseMap` as an argument to the function, if it is not used externally?
seems like only the `RedisResponse` is used

Reply:
`RedisResponseMap` is simply a typedef for `boost::redis::response<std::map<std::string, std::string>>`

Reply:
my question was why do we need it as a parameter to the function, if it is no used outside the function?

Reply:
It is used outside of the `doRedisFunc` by the function that called it. 
It stores the response to be used by the calling function as desired. 

Since it can be different for different functions, we let them handle its return values.

Reply:
can you please point to where it is being used?
also, whay do we need both `RedisResponseMap` and `RedisResponse`?